src/hotspot/share/opto/regmask.hpp

```diff
@@ -131,76 +401,85 @@ class RegMask {
     assert(valid_watermarks(), "post-condition");
   }
 
-  // Handy copying constructor
-  RegMask(RegMask *rm) {
-    _hwm = rm->_hwm;
-    _lwm = rm->_lwm;
-    for (unsigned i = 0; i < _RM_SIZE; i++) {
-      _RM_UP[i] = rm->_RM_UP[i];
-    }
+  // Construct an empty mask
+  RegMask(Arena* arena)
+      : _RM_UP(), _rm_size(_RM_SIZE), _offset(0), _all_stack(false),
+        _lwm(_RM_MAX), _hwm(0), _arena(arena) {
     assert(valid_watermarks(), "post-condition");
   }
-
-  // Construct an empty mask
-  RegMask() : _RM_UP(), _lwm(_RM_MAX), _hwm(0) {
+  RegMask() : RegMask(nullptr) {
     assert(valid_watermarks(), "post-condition");
   }
 
   // Construct a mask with a single bit
-  RegMask(OptoReg::Name reg) : RegMask() {
+  RegMask(OptoReg::Name reg,
+          Arena* arena
+          DEBUG_ONLY(COMMA bool orig_const = false))
+      : RegMask(arena) {
     Insert(reg);
+    DEBUG_ONLY(this->orig_const = orig_const;)
```

Comment:
Could this not be part of the initializer list? Or does the `Insert` prevent that?

Reply:
I moved things around a bit to make sure it can be put in the initializer list. I couldn't before because the initializer list contains a delegating constructor.